### DISCUSSION TITLE
How to add token in header as Authorization in client side when using "credentials"?

### DISCUSSION THREAD

**Discussion by mossen** [nextauthjs/next-auth#1703] [Help]
Trying to add the token into header from client side, by using session/useSession one request is sent to backend which is not ideal.

Wanna add token in the header just only from client side or if there is a way to attach the token automatically into the header as `Authorization`.

**Comment by balazsorban44** [comment]
Could you please give more context? Do you want the access token available in the session?
check the jwt and session callbacks:
https://next-auth.js.org/configuration/callbacks

**Reply by mossen** [reply]
I am using "credentials" as provider and a third party to get some services, this third party provides a user token which needs to be sent in header as "Authorization" after user logs in.
I am using server side to send the login API request to not exposing API key and secret key but after getting the user token all the APIs can be sent from client side.
Is there anyway to use this third party user token without sending a `/api/auth/session` request?
Currently, saving this user token into cookies after logging in the user manually and using it.
